tests/pose_estimation_pytorch/data/test_transforms.py

```diff
@@ -101,3 +103,213 @@ def test_coarse_dropout():
+            ),
+        },
+        {
+            "image_shape": [480, 640, 3],
+            "transform_config": dict(
+                shift_factor=0.0,
+                shift_prob=0.0,
+                scale_factor=[0.5, 1.25],
+                scale_prob=1.0,
+            ),
+        },
+        {
+            "image_shape": [480, 640, 3],
+            "transform_config": dict(
+                shift_factor=0.0,
+                shift_prob=0.0,
+                scale_factor=[0.5, 1.5],
+                scale_prob=0.5,
+            ),
+        },
+    ],
+)
+def test_random_bbox_transform_scale(data: dict) -> None:
+    _set_random_seed()
+
+    h, w, c = data["image_shape"]
+    image_wh = np.array([w, h])
+
+    gen = np.random.default_rng(seed=0)
+
+    # generate 100 bboxes
+    num_bboxes = 100
+    bboxes = np.zeros((num_bboxes, 4))
+    # sample x, y in the images
+    bboxes[:, :2] = image_wh * gen.random((num_bboxes, 2))
+    # sample w, h with the space remaining
+    bboxes[:, 2:] = (image_wh - bboxes[:, :2]) * gen.random((num_bboxes, 2))
+
+    print()
+    print("Input bounding boxes")
+    print(bboxes)
+
+    t = A.Compose(
+        [transforms.RandomBBoxTransform(**data["transform_config"])],
+        bbox_params=A.BboxParams(format="coco", label_fields=["bbox_labels"]),
+    )
+    output = t(
+        image=np.zeros((h, w, c)), bboxes=bboxes, bbox_labels=np.zeros(len(bboxes)),
+    )
+    print("Output bounding boxes")
+    for out_bbox in output["bboxes"]:
+        print(out_bbox)
+    print()
+
+    bboxes_out = np.asarray(output["bboxes"])
+    scale_low, scale_high = data["transform_config"]["scale_factor"]
+    for bbox_in_wh, bbox_out_wh in zip(bboxes[:, 2:], bboxes_out[:, 2:]):
+        print("bbox_in_wh", bbox_in_wh)
+        w, h = bbox_in_wh[0].item(), bbox_in_wh[1].item()
+        w_low, w_high = w * scale_low, w * scale_high
+        h_low, h_high = h * scale_low, h * scale_high
+        print("(w, w_low, w_high)", w, w_low, w_high)
+        print("(h, h_low, h_high)", h, h_low, h_high)
+        assert w_low <= bbox_out_wh[0].item() <= w_high
+        assert h_low <= bbox_out_wh[1].item() <= h_high
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "image_shape": [480, 640, 3],
+            "transform_config": dict(
+                shift_factor=0.1,
+                shift_prob=1.0,
+                scale_factor=[1.0, 1.0],
+                scale_prob=0.0,
+            ),
+        },
+    ],
+)
+def test_random_bbox_transform_shift(data: dict) -> None:
+    _set_random_seed()
+
+    h, w, c = data["image_shape"]
+    image_wh = np.array([w, h])
+
+    gen = np.random.default_rng(seed=0)
+
+    # generate 100 bboxes
+    num_bboxes = 100
+    bboxes = np.zeros((num_bboxes, 4))
+    # sample x, y in the images
+    bboxes[:, :2] = image_wh * gen.random((num_bboxes, 2))
+    # sample w, h with the space remaining
+    bboxes[:, 2:] = (image_wh - bboxes[:, :2]) * gen.random((num_bboxes, 2))
```

Comment:
The bboxes generation is duplicated across tests, let's move this out into a separate method?